src/app/server/Mdns.h

```diff
@@ -19,26 +19,37 @@
 
 #include <core/CHIPError.h>
 #include <stddef.h>
+#include <support/BitFlags.h>
 
 namespace chip {
 namespace app {
 namespace Mdns {
 
+enum class CommissioningMode
+{
+    kDisabled,
+    kEnabled,
+    kEnabledAsAdditionalCommissioning
+};
+
 /// Start operational advertising
 CHIP_ERROR AdvertiseOperational();
 
 /// Set MDNS commissioner advertisement
 CHIP_ERROR AdvertiseCommissioner();
 
 /// Set MDNS commissionable node advertisement
-CHIP_ERROR AdvertiseCommissionableNode();
+CHIP_ERROR AdvertiseCommissionableNode(CommissioningMode mode);
 
-/// Set MDNS advertisement
-// CHIP_ERROR Advertise(chip::Mdns::CommssionAdvertiseMode mode);
-CHIP_ERROR Advertise(bool commissionableNode);
+/// (Re-)starts the minmdns server using default state
+// Commissioning mode is enabled if device has not yet been commissioned
+void StartServer();
 
 /// (Re-)starts the minmdns server
-void StartServer();
+//
+// NOTE: when device has never been commissioned, kAdditionalCommissioning flag will be ignored
```

Comment:
Confusing comment, because there is no such flag....  Please have this refer to the actual value that will be "ignored" and say what "ignored" means: will we not advertise commissioning mode at all (because someone is just passing random values in), or will we advertise it but without the AC=1/_A bits?   I'd somewhat prefer the former....

Reply:
cleaned up